system variable: add tidb_opt_skew_distinct_agg (#9681)
pingcap/docs: system-variables.md

3 rows in set (0.00 sec)
```

### tidb_opt_skew_distinct_agg <span class="version-mark">New in v6.2.0</span>

> 

Author: ```suggestion

> **Note:**
>
> The optimization enabled by this variable is effective **only for TiFlash**.

```

- Scope: SESSION | GLOBAL

> - Persists to cluster: No

Reviewer: Should be YES. The Chinese version needs to be updated too.

Author: ```suggestion
- Persists to cluster: Yes
```

- Type: Boolean
- Default value: `OFF`

> - This variable sets whether the optimizer rewrites the aggregate functions with `DISTINCT` to the two-level aggregate functions, such as rewriting `SELECT b, COUNT(DISTINCT a) FROM t GROUP BY b` to `SELECT b, COUNT(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`. When the aggregation column has serious skew and the `DISTINCT` column has many different values, this rewriting can avoid the data skew in the query execution and improve the query performance. This optimization is effective **only for TiFlash**.

Author: ```suggestion
- This variable sets whether the optimizer rewrites the aggregate functions with `DISTINCT` to the two-level aggregate functions, such as rewriting `SELECT b, COUNT(DISTINCT a) FROM t GROUP BY b` to `SELECT b, COUNT(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`. When the aggregation column has serious skew and the `DISTINCT` column has many different values, this rewriting can avoid the data skew in the query execution and improve the query performance. 
```

> - This variable sets whether the optimizer rewrites the aggregate functions with `DISTINCT` to the two-level aggregate functions, such as rewriting `SELECT b, COUNT(DISTINCT a) FROM t GROUP BY b` to `SELECT b, COUNT(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`. When the aggregation coloumn has serious skew and the `DISTINCT` column has many different values, this rewriting can avoid the data skew in the query execution and improve the query performance. This optimization is effective **only for TiFlash**.

Reviewer 2: ```suggestion
- This variable sets whether the optimizer rewrites the aggregate functions with `DISTINCT` to the two-level aggregate functions, such as rewriting `SELECT b, COUNT(DISTINCT a) FROM t GROUP BY b` to `SELECT b, COUNT(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`. When the aggregation column has serious skew and the `DISTINCT` column has many different values, this rewriting can avoid the data skew in the query execution and improve the query performance. This optimization is effective **only for TiFlash**.
```

3 rows in set (0.00 sec)
```

### tidb_opt_skew_distinct_agg <span class="version-mark">New in v6.2.0</span>

> **Note:**
>

> > The optimization enabled by this variable is effective **only for TiFlash**.

Author: ```suggestion
> The query performance optimization by enabling this variable is effective **only for TiFlash**.
```

3 rows in set (0.00 sec)
```

### tidb_opt_skew_distinct_agg <span class="version-mark">New in v6.2.0</span>

- Scope: SESSION | GLOBAL
- Persists to cluster: No

> - Default valiue: `OFF`

Reviewer 3: ```suggestion
- Type: Boolean
- Default value: `OFF`
```

> - This variable sets whether the optimizer rewrites the aggregate functions with `DISTINCT` to the two-level aggregate functions, such as rewriting `SELECT b, count(DISTINCT a) FROM t GROUP BY b` to `SELECT b, count(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`. When the aggregation coloumn has serious skew and the `DISTINCT` column has many different values, this rewriting can avoid the data skew in the query execution process and improve the query performance. This optimization is effective **only for TiFlash**.

Reviewer 3: ```suggestion
- This variable sets whether the optimizer rewrites the aggregate functions with `DISTINCT` to the two-level aggregate functions, such as rewriting `SELECT b, COUNT(DISTINCT a) FROM t GROUP BY b` to `SELECT b, COUNT(a) FROM (SELECT b, a FROM t GROUP BY b, a) t GROUP BY b`. When the aggregation coloumn has serious skew and the `DISTINCT` column has many different values, this rewriting can avoid the data skew in the query execution and improve the query performance. This optimization is effective **only for TiFlash**.
```
